client/state/plugins/marketplace/actions.ts

```diff
@@ -40,3 +67,168 @@ export function setIsPluginInstalledDuringPurchase(
 		isPluginInstalledDuringPurchase,
 	};
 }
+
+export function resetPurchaseFlow(): { type: string } {
+	return {
+		type: MARKETPLACE_RESET_PURCHASE_FLOW,
+	};
+}
+
+export function siteTransferStateChange(
+	state: MARKETPLACE_ASYNC_PROCESS_STATUS,
+	reason = ''
+): { type: string; state: MARKETPLACE_ASYNC_PROCESS_STATUS; reason: string } {
+	return {
+		type: MARKETPLACE_SITE_TRANSFER_STATE_CHANGE,
+		state,
+		reason: reason ?? 'NOT_PROVIDED',
```

Comment:
What's the intended usage of this `reason` field? From the usage below, it looks like a human readable error message. Are they for users or developers? If they are for users, they will need to be localized. If they are for developers, why it is not simply a status code which can be more compact?

The same question applies to all the `reason` field below :)

Reply:
Since this is a very chaotic flow I thought of the need to have a way to mention  a message to pin point where it originated from. Wanted it to be more human readable since it will be easier to just explain why a particular state changed (error states most importantly) are triggered.